### ISSUE TITLE
✨ Pool: Understanding classifications

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#8831]
## ✨ Feature

Pools are getting a new "Understanding classifications" field.

## 🎨 Design File

@substrae 

Pool editor:
https://www.figma.com/file/e2DTvyppj41m6slFWTDZ0U/Admin---Job-advertisement-creation?type=design&node-id=52-40341&mode=design&t=cs4ucOAt8RuZ44or-0

Pool advertisement:
https://www.figma.com/file/CdYmWlP6SpT17bw7Q9411Z/Candidates---Job-advertisement?type=design&node-id=260-5309&mode=design&t=83ZYhqDekaDoFFyC-0

## 📸 Screenshot

Pool editor:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/92d44a00-3649-4108-8b5d-645c9839a154)

@substrae Looks like the editable form design is not using the editable input fields?

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/eb8fbaf2-1a11-4147-bf97-a8a067ad812f)

Pool advertisement
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/a5263406-da49-4944-8af8-bb519a4ba965)

## ✅ Acceptance Criteria

- [ ] migration, database field, Eloquent model
- [ ] seeder, factor updates
- [ ] Graphql query update, mutation update
@substrae Should we use the existing classification descriptions as the new templates?
- [ ] Pool editor, editable state
  - [ ] In instances where a classification's level is high enough for managerial roles (IT-03 thru IT-05), allow the user to select from descriptions for Team lead, Advisor, or Both, which then provides the relevant text
- [ ] Pool editor, readonly state
- [ ] Pool advertisement
- [ ] frontend "pool can publish" validation update
- [ ] backend "pool can publish" validation update

**Comment by tristan-orourke** [comment]
@substrae So to be clear, what we're adding to the data model some way for pools to distinguish whether its for leadership roles or not (or whether the pool can collect both). But only certain classifications need this distinction? And the copy that appears depends on the classification as well. Do we have all that copy ready to go?

**Comment by substrae** [comment]
@tristan-orourke 
Yes - each classification/level combo might contain a single role or two roles, representing management vs. individual contributor. In `IT`'s case, IT-01 and 02 have a single role, while IT-03, 04, and eventually 05 contain the two options. The text used should just be the existing text (it currently exists within two distinct `details` components).

**Comment by gobyrne** [comment]
Lauren is hoping this will be sufficiently addressed by the job poster templates work being done. There will be dedicated stream (or sub-streams) that will separate these out.

Given that, and it's not a major pain point right now, I'm gonna ice this.

**Comment by petertgiles** [comment]
No movement.  Reopen if we want to start working on it.
